packages/grafana-llm-app/pkg/mcp/live_server.go

```diff
@@ -94,14 +124,42 @@ type liveSession struct {
 	// sender is the StreamSender for the Grafana Live session. It is used to send
 	// JSON-RPC responses back to the client.
 	sender *backend.StreamSender
+
+	// accessToken is the access token for the Grafana Live session.
+	accessToken string
 }
 
 // HandleStream handles a new Grafana Live session.
 func (s *GrafanaLiveServer) HandleStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
+	var tokenExchangeResponse *authn.TokenExchangeResponse
+	// Only do the token exchange if we are in Grafana Cloud.
+	if s.EnableGrafanaManagedLLM {
+		var tokenExchangeClient *authn.TokenExchangeClient
+		var err error
+		// We need to use the token exchange API to create a token for the Grafana Live session.
+		tokenExchangeClient, err = authn.NewTokenExchangeClient(authn.TokenExchangeConfig{
+			Token:            s.LLMAppAccessPolicyToken,
+			TokenExchangeURL: "http://api-lb.auth.svc.cluster.local./v1/sign-access-token", // TODO: make this configurable.
+		})
+		if err != nil {
+			return err
+		}
+
+		tokenExchangeResponse, err = tokenExchangeClient.Exchange(ctx, authn.TokenExchangeRequest{
+			Namespace: fmt.Sprintf("stack-%s", s.Tenant),
+			Audiences: []string{"grafana"},
+			ExpiresIn: &[]int{600}[0], // this is 10 minutes
```

Comment:
nit: You don't need to provide this if you're OK with the 10 minute expiration which is set by default.